src/db/oracledb/books-dao.js

```diff
@@ -0,0 +1,130 @@
+import _ from 'lodash';
+
+import oracledb from 'oracledb';
+
+import { parseQuery } from 'utils/parse-query';
+
+import { getConnection } from './connection';
+
+
+
+/**
+ * Return a list of books
+ *
+ * @param {object} query Query parameters
+ * @returns {Promise<object>[]} Promise object represents a list of books
+ */
+const getBooks = async (query) => {
+  const connection = await getConnection();
+  try {
+    const parsedQuery = parseQuery(query); // contrib is changed
+    const { rawBooks } = await connection.execute(parsedQuery);
+    return rawBooks;
+  } finally {
+    connection.close();
+  }
+};
+
+/**
+ * Return a specific book by unique ID
+ *
+ * @param {string} id Unique book ID
+ * @returns {Promise<object>} Promise object represents a specific book or return undefined if term
+ *                            is not found
+ */
+const getBookById = async (id) => {
+  const connection = await getConnection();
+  const query = `
+  SELECT *
+  FROM library_api_books
+  WHERE book_id = :bookId
+  `;
+
+  try {
+    const bindVars = {
+      bookId: id
+    };
+
+    const { rows } = await connection.execute(query, bindVars);
+    if (_.isEmpty(rows)) {
+      return undefined;
+    }
+    if (rows.length > 1) {
+      throw new Error('Expect a single object but got multiple results.');
+    } else {
+      const rawBooksCaps = rows[0];
+      const rawBooks = Object.keys(rawBooksCaps).reduce((result, key) => {
+        result[key.toLowerCase()] = rawBooksCaps[key];
+        return result;
+      }, {});
+      return rawBooks;
+    }
+  } finally {
+    connection.close();
+  }
+};
+
+/**
+ * Posts a new book to the Oracle database
+ *
+ * Inserts the posted book into the Oracle database.
+ *
+ * @param {object} body Request body
+ * @returns {Promise<object>} Promise object represents the posted book
+ */
+const postBook = async (body) => {
+  const connection = await getConnection();
+
+  try {
+    const newBookData = body.data.attributes;
+    newBookData.available = 'true';
```

Comment:
You don't have to assign it to `newBookData` here, instead, you can directly set it to true in `bindVars`.

Reply:
I had done that previously but then changed it, because it helps when I return the newBookData object at the end of the function, that object is filled with all the attributes needed for response.